### PR TITLE
Dat257 71 refactor fluxor

### PR DESCRIPTION
**Pull request by Flaskhals68**
This will add state manegment for the custom compare page.
This will also fix a bug on the home page that removes the filter component when no data is selected.

### CODE REVIEW

File: Client/Components/DashBoard.razor [hunk 1]
```diff
@@ -1,14 +1,14 @@
 ﻿@inherits Fluxor.Blazor.Web.Components.FluxorComponent;
 
 <div class="dash-board">
-    @if (!State.Value.HomeCountryFound)
+    @if (_homeCountryError)
```
Comment (filipkauffeldt) [inline Client/Components/DashBoard.razor:4]: Why not use the state?

File: Client/Components/DashBoard.razor.cs [hunk 2]
```diff
@@ -100,18 +97,20 @@ private async Task InitHomeCountryAsync()
...
-                var country = await apiHandler.FetchHomeCountryAsync(httpClient);
+                var country = await _apiHandler.FetchHomeCountryAsync(_httpClient);
                    
                 if (country != null)
                 {
                     Dispatcher.Dispatch(new HomeCountryDetectedSuccessfullyAction(country));
                 }
                 else
                 {
+                    _homeCountryError = true;
```
Comment (filipkauffeldt) [inline Client/Components/DashBoard.razor.cs:108]: Why do we need this over using the state?

File: Client/Store/Actions/CustomCompareActions.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace Client.Store.Actions
+{
+    public class OriginCountryChosenAction
+    {
+        public Country Country { get; }
+        public OriginCountryChosenAction(Country country) => Country = country;
+    };
+
+    public class OriginCountryChoiceFailedAction { };
```
Comment (filipkauffeldt) [inline Client/Store/Actions/CustomCompareActions.cs]: Why do these actions have differently structured names?

**Review by filipkauffeldt — COMMENTED** [review]
Looks good for the most part

File: Client/Components/DashBoard.razor (hunk 1 above)
Comment (Flaskhals68) [reply Client/Components/DashBoard.razor:4]: Using the HomeCountryFound causes the error message for when the user's home country couldn't be detected to briefly display. Using the internal _homeCountryError prevents this. 

You could probably use a HomeCountryError property stored in the state instead to solve the problem.
Would that be preffered instead?

File: Client/Store/Actions/CustomCompareActions.cs (hunk 3 above)
Comment (Flaskhals68) [reply Client/Store/Actions/CustomCompareActions.cs]: Removed the unused action. The error should probably be handled outside of the state.

File: Client/Components/DashBoard.razor (hunk 1 above)
Comment (filipkauffeldt) [reply Client/Components/DashBoard.razor:4]: Not sure. If this works it should be fine

**Review by filipkauffeldt — APPROVED** [review]

**Pull request merged**